libqtile/layout/wmii.py

```diff
@@ -0,0 +1,458 @@
+# SOFTWARE.
+
+from __future__ import division
+
+from .base import Layout
+
+# This layout implements something akin to wmii's semantics.
+#
+# Each group starts with one column.  The first window takes up the whole screen.
+# Next window splits the column in half.  Windows can be moved to the column to
+# the left or right.  If there is no column in the direction being moved into, a
+# new column is created.
+#
+# Each column can be either stacked (each window takes up the whole vertical real
+# estate) or split (the windows are split equally vertically in the column)
+# Columns can be grown horizontally (cmd_grow_left/right).
+#
+# My config.py as the following added:
+
+#    Key(
+#        [mod, "shift", "control"], "l",
+#        lazy.layout.grow_right()
+#    ),
+#    Key(
+#        [mod, "shift"], "l",
+#        lazy.layout.shuffle_right()
+#    ),
+#    Key(
+#        [mod, "shift", "control"], "h",
+#        lazy.layout.grow_left()
+#    ),
+#    Key(
+#        [mod, "shift"], "h",
+#        lazy.layout.shuffle_left()
+#    ),
+#    Key(
+#        [mod], "s",
+#        lazy.layout.toggle_split()
+#    ),
+
+# We have an array of columns.  Each columns is a dict containing
+# width (in percent), rows (an array of rows), # and mode, which is
+# either 'stack' or 'split'
+#
+# Each row is an array of clients
+
+def swap(array, a, b):
+    tmp = array[b]
+    array[b] = array[a]
+    array[a] = tmp
+
+class Wmii(Layout):
+    """
+        This layout emulates wmii layouts.  The screen it split into
+        columns, always starting with one.  A new window is created in
+        the active window's column.  Windows can be shifted left and right.
+        If there is no column when shifting, a new one is created.
+        Each column can be stacked or divided (equally split).
+    """
+    defaults = [
+        ("border_focus", "#881111", "Border colour for the focused window."),
+        ("border_normal", "#220000", "Border colour for un-focused windows."),
+        ("border_focus_stack", "#0000ff", "Border colour for un-focused windows."),
+        ("border_normal_stack", "#000022", "Border colour for un-focused windows."),
+        ("grow_amount", 5, "Amount by which to grow/shrink a window."),
+        ("border_width", 2, "Border width."),
+        ("name", "wmii", "Name of this layout."),
+        ("margin", 0, "Margin of the layout"),
+    ]
+
+    def __init__(self, **config):
+        Layout.__init__(self, **config)
+        self.add_defaults(Wmii.defaults)
+        self.current_window = None
+        self.clients = []
+        self.columns = [{'active': 0, 'width': 100, 'mode': 'split', 'rows': []}]
+
+    def info(self):
+        d = Layout.info(self)
+        d["current_window"] = self.current_window
+        d["clients"] = [x.name for x in self.clients]
+        return d
+
+    def add_column(self, prepend, win):
+        newwidth = (int)(100 / (len(self.columns) + 1))
```

Comment:
`int` is a function, not a cast, so this can be `int(100 / (len(self.columns) + 1))`